CGPs/0013.md

```diff
@@ -0,0 +1,83 @@
+
+Changes that are being proposed in this release are described in [this Github release](https://github.com/celo-org/celo-monorepo/releases/tag/celo-contracts-v1.rc1). OpenZeppelin audited all core Celo contracts and found no “critical” or “high” severity issues. The audit report can be found [here](https://blog.openzeppelin.com/celo-contracts-audit/#phase-3).
+
+## Proposed Changes
+
+This Celo Core Contracts Upgrade proposal is special since it operationalizes the new [release process](https://docs.celo.org/community/release-process/smart-contracts). As part of the new versioning schema, most Celo Core Contracts added a `getVersionNumber` function which changed the bytecode of all implementations. For that reason, this proposal contains transactions to update the implementation contract of these contracts' proxies.
+
+1. AccountsProxy `setImplementation`
+2. AttestationsProxy `setImplementation`
+3. DoubleSigningSlasherProxy `setImplementation`
+4. ElectionProxy `setImplementation`
+5. EpochRewardsProxy `setImplementation`
+6. EscrowProxy `setImplementation`
+7. ExchangeProxy `setImplementation`
+8. GasPriceMinimumProxy `setImplementation`
+9. GoldTokenProxy `setImplementation`
+10. GovernanceProxy `setImplementation`
+11. LockedGoldProxy `setImplementation`
+12. RandomProxy `setImplementation`
+13. ReserveProxy `setImplementation`
+14. SortedOraclesProxy `setImplementation`
+15. StableTokenProxy `setImplementation`
+16. UsingPrecompilesProxy `setImplementation`
+17. ValidatorsProxy `setImplementation`
+
+Additionally, to avoid future updated bytecode of Celo Core Contracts when underlying libraries are changed, some libraries are being proxied and linked via the proxy. That necessitates the following transactions:
+
+1. Registry `setAddressFor` Signatures library
+2. Registry `setAddressFor` AddressLinkedList library
+3. Registry `setAddressFor` AddressSortedLinkedList library
+4. Registry `setAddressFor` AddressSortedLinkedListWithMedian library
+5. Registry `setAddressFor` Proposals library
+6. Registry `setAddressFor` IntegerSortedLinkedList library
+
+Regular features that were being added or bugs that were fixed are described in the [Github release](https://github.com/celo-org/celo-monorepo/releases/tag/celo-contracts-v1.rc1). Since the major version of these Celo Core Contracts changed due to the version change, no separate transactions will reflect the "regular" changes. One exception to that is the newly added `DowntimeSlasher`:
+
+1. Registry `setAddressFor` DowntimeSlasher
+
+Since this release is larger than usual and more opaque, it is crucial for Celo stakeholders to follow the verification steps below.
+
+## Verification
```

Comment:
Should this section also include instructions for verifying the current deployment as well as that versions have been correctly set between the current deployment and the proposed release?

Reply:
I had tried to keep it short as to not effectively copy everything from the docs. I'd also imagine that stakeholders will mostly care about verifying the proposal, than verifying the current deployment (as there is nothing this proposal could do about it)

Reply:
Though I guess the version check is maybe a good idea